### DISCUSSION TITLE
Behavior function return type

### DISCUSSION THREAD

**Discussion by brunohoplite** [StateSmith/StateSmith#237] [Q&A] [answered]
Is there a way to specify the return type of the bahvior function (event handler function type) and the dispatch function as well?
So instead of having `typedef void (*MY_SM_NAME_Func)(MY_SM_NAME* sm);` and  `void MY_SM_NAME_dispatch_event(MY_SM_NAME* sm, MY_SM_NAME_EventId event_id);` we could have something like

`typedef int (*MY_SM_NAME_Func)(MY_SM_NAME* sm);` and `int MY_SM_NAME_dispatch_event(MY_SM_NAME* sm, MY_SM_NAME_EventId event_id);`

This would be really helpful beause my code follows the error return pattern and the functions called by the state-machine return error codes that ideally have to be handled.

Thank you!
Bruno

**Comment by adamfk** [comment, marked as answer]
That would be helpful. I started working on it a while ago, but it will
still a good while before I find time to finish it. I actually need it for
supporting orthogonal states eventually.

In the meantime, could you use a state machine variable to store the result?

**Reply by brunohoplite** [reply]
That's excellent news!
And yes, storing the result in a state-machine variable is a good workaround for now.

Thank you for the suggestion and prompt response :)

**Reply by adamfk** [reply]
You're welcome! I'm always curious to know what people are working on. Any high level details you can share? C, C++, C#, JS?

**Reply by brunohoplite** [reply]
I mostly work on bare-metal embedded devices in C at work and in my hobby projects. StateSmith is THE tool that I've been looking for and has been really helpful and fun to work with.
